src/mdo/java/WrapperProperties.java

```diff
@@ -208,16 +182,19 @@ interface WriteOpVoid {
     }
 
     private <T> T writeOperation(WriteOp<T> runner) {
-        ensureInitialized();
-        T ret = runner.perform(orderedProps);
-        setter.accept(orderedProps);
+        OrderedProperties props = new OrderedProperties(getter.get());
+        T ret = runner.perform(props);
+        if (!props.equals(getter.get())) {
+            setter.accept(props);
+        }
         return ret;
     }
 
     private void writeOperationVoid(WriteOpVoid runner) {
-        ensureInitialized();
-        runner.perform(orderedProps);
-        setter.accept(orderedProps);
+        writeOperation(properties -> {
+            runner.perform(properties);
+           return null;
```

Comment:
```suggestion
           runner.perform(properties);
           return null;
```
format seems broken as offset.

Reply:
yes, thanks.